### ISSUE TITLE
Why GRANDE for regression doesnt outperform Catboost and XGB

### ISSUE THREAD

**Issue by waminos** [s-marton/GRANDE#3]
Hi ! thanks for your great work.
As the title says it I read the paper and test GRANDE on my data
and it seems the results doesn't outperform catboost.
I also seen that's the same conclusion on your minimal example.
Any idea how to improve or why it doesn't outperform  as classification ?
Thanks :)

**Comment by s-marton** [comment]
Hey @waminos,
thank you for your message and the interest in our method!

As stated in our paper, there is no universal method for tabular data. Consequently, there exist datasets where the performance of GRANDE is not that strong. Nevertheless, it is important to note that on several regression datasets, GRANDE archives superior performance as well (see for instance the updated minimal example for regression on the dataset_sales dataset).

Indeed, when compared with GBDTs such as CatBoost and XGBoost, the average performance of GRANDE on regression tasks is not as strong as it is on classification tasks. However, this is not unique to GRANDE, as the overall gap between gradient-based methods and GBDTs tends to be larger for regression tasks compared to classification tasks.

For instance, the arbitrary target scale of regression tasks can be problematic for gradient-based methods. Currently, this is solved by normalizing the target, but there might be more optimal solutions depending on the task.

Additionally, we did not focus on optimizing GRANDE for regression tasks until now, but this is something we are planning to look into. As soon as we have updates, we will include them in the repo and let you know. :-)

**Comment by waminos** [comment]
Thanks for answer ! I will continue to experiment and keep an eye if an update comes.
Cheers

**Comment by black940514** [comment]
Could you share more details about why GRANDE doesn't outperform CatBoost/XGBoost on regression tasks? Are there specific hyperparameters that need tuning for regression?

**Comment by s-marton** [comment]
I think that with the latest updates and the use of categorical and numerical embeddings, regression performance should already have improved (specifically when tuning the use of embeddings and their learning rate as hyperparameters). However, compared to classification, regression still falls behind. I am currently investigating this and hope to provide an update with more information soon.
